pkg/config/settings.go

```diff
@@ -32,3 +39,86 @@ const (
 
 	NodeAgentPort = "10086"
 )
+
+var ReplaceControllerVersions = sets.String{
+	ControllerRepository + ":" + "1.0.0": sets.Empty{},
+}
+
+var ReplaceNodeVersions = sets.String{
+	NodeRepository + ":" + "1.0.0": sets.Empty{},
+}
+
+var ReplaceCSIProvisionerVersions = sets.String{
+	"quay.io/k8scsi/csi-provisioner:v1.3.0": sets.Empty{},
+}
+
+var ReplaceCSIAttacherVersions = sets.String{}
+
+var ReplaceNodeDriverRegistrarVersions = sets.String{}
+
+var ReplaceLivenessProbeVersions = sets.String{}
+
+var ReplaceOpenShiftControllerVersions = sets.String{
+	OpenShiftControllerRepository + ":" + "1.0.0": sets.Empty{},
+}
+
+var ReplaceOpenShiftNodeVersions = sets.String{
+	OpenShiftNodeRepository + ":" + "1.0.0": sets.Empty{},
+}
+
+var ReplaceOpenShiftCSIProvisionerVersions = sets.String{
+	"quay.io/k8scsi/csi-provisioner:v1.3.0": sets.Empty{},
```

Comment:
these should come from consts

Reply:
ReplaceOpenShiftCSIProvisionerVersions is also a const

Reply:
then I'm not sure I follow - what happens if in CSI 1.2.0 we upgrade our provisioner to 1.5.0? do we need both 1.3.0 and 1.4.0 here? where is the version comparison?

Reply:
yes, both 1.3 and 1.4.
What do you mean by version comparison?

Reply:
I thought there might be missing some version comparison logic between current and latest version, but I guess new CSI versions want the option to also downgrade sidecars in the future.

I still don't like the idea of having to add each used version to a set (how will the maintain even know to do it?).
I still think it should come from the OLM directory, all the info is already there, and if the `alm-examples` format will change, we will add parsing support for it here.

Reply:
This is exactly the opposite of CSI-1055 purpose

Reply:
1. we do not support downgrade, if the user use an old image in new version, wew can not guarantee the availability.
2. I think it is part of the upgrade procedure (replacing old image with new one), and it is easy to understand. Don't worry you will forget it, we have test cases to cover it.
3. I still think the code is the source, the examples should come from the code in some way.
4. Sorry, I don't understand CSI-1055, there is no context.

Reply:
1. I didn't mean the user choose to use an old image, I meant that we want the option to officially downgrade a sidecar in a new CSI release.
2. I don't think it's easy for the user to know exactly which old images we decide to replace or not, since they shouldn't be aware of our entire versions history.
3. If you know of a way to generate the examples from the code, that will also be fine.
4. The purpose of CSI-1055 is to reduce the amount of places where we maintain the images and versions, ideally to one location (this PR currently does the opposite).